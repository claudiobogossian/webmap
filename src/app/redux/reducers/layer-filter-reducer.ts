import { createReducer, createAction, on, props } from '@ngrx/store'

/* ACTIONS / CREATORS */
export const actions = {
  setInitialDate: createAction('[FILTER] set initial date', props<{initialDate: Date}>()),
  setFinalDate: createAction('[FILTER] set final date', props<{finalDate: Date}>())
}
export const featureKey = 'layerFilter'

export interface State {
  initialDate: Date;
  finalDate: Date;
}

const initialState: State = {
  initialDate: null,
  finalDate: null
}

/* REDUCER */
export const reducer = createReducer(
  initialState,
  // Even thought the `state` is unused, it helps infer the return type
  on(actions.setFinalDate, (state, initialDate: any) => ({ ...state, initialDate })),
  on(actions.setInitialDate, (state, finalDate: any) => ({ ...state, finalDate }))
)

// SELECTOR
export const selectInitialDate = (state: State) => state.initialDate
export const selectFinalDate = (state: State) => state.finalDate