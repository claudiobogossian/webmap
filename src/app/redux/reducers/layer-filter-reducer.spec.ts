import { actions, featureKey, reducer } from './layer-filter-reducer'

describe('LayerFilterReducer', () => {
  describe('ACTIONS:', () => {
    it('feature key should be "layerFilter"', () => {
      expect(featureKey).toBe('layerFilter')
    })

    it('setInitialDate should be a function', () => {
      expect(actions.setInitialDate).toBeInstanceOf(Function)
    })

    it('setFinalDate should be a function', () => {
      expect(actions.setFinalDate).toBeInstanceOf(Function)
    })
  })

  describe('REDUCERS', () => {
    it('should return the default state', () => {
      const result = reducer(undefined, {} as any)
      expect(result).toMatchSnapshot()
    })

    // it('should dispatch the setInitialDate and return the merged object', () => {
    //   const action = actions.setInitialDate({ date: new Date() })
    //   const result = reducer(fromBooks.initialState, action)
    //     console.log('====================================');
    //     console.log({result});
    //     console.log('====================================');
    //   //   expect(result).toMatchSnapshot()
    // })

    it('should dispatch the setFinalDate and return the merged object', () => {})
  })

  describe('SELECTORS', () => {})
})
