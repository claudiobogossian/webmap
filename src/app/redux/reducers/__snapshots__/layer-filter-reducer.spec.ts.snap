// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayerFilterReducer REDUCERS should return the default state 1`] = `
Object {
  "finalDate": null,
  "initialDate": null,
}
`;
